src/database/memory.rs

```diff
@@ -512,13 +513,29 @@ macro_rules! populate_test_db {
         };
 
         let txid = tx.txid();
-        let confirmation_time = tx_meta
+        // Set Confirmation time only if current height is provided. Else None.
+        let confirmation_time = if let Some(height) = current_height {
+            tx_meta
             .min_confirmations
             .and_then(|v| if v == 0 { None } else { Some(v) })
             .map(|conf| $crate::BlockTime {
-                height: current_height.unwrap().checked_sub(conf as u32).unwrap() + 1,
+                height: height.checked_sub(conf as u32).unwrap() + 1,
                 timestamp: 0,
-            });
+            })
+        } else {
+            None
+        };
```

Comment:
I don't think it's ok to silently drop the `min_confirmations` if the current height is not provided. I would rather keep what we had before with a better error message, by changing the `unwrap()` into an `expect()`.

Reply:
Makes sense..